src/main/java/ezschedule/ui/Calendar.java

```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * Creates a {@code Calender} with the given {@code ObservableList}{@code CommandExecutor}}.
+     */
+    public Calendar(ObservableList<Event> eventList, FilterExecutor filterExecutor) {
+        super(FXML);
+        this.eventList = eventList;
+        this.filterExecutor = filterExecutor;
+        date = ZonedDateTime.now();
+        today = ZonedDateTime.now();
+        monthMaxDate = date.getMonth().maxLength();
+        eventList.addListener((ListChangeListener<Event>) c -> refreshCalendar());
+        drawCalendar();
+    }
+
+    @FXML
+    void backOneMonth() {
+        date = date.minusMonths(1);
+        monthMaxDate = date.getMonth().maxLength();
+        calendar.getChildren().clear();
+        drawCalendar();
+    }
+
+    @FXML
+    void forwardOneMonth() {
+        date = date.plusMonths(1);
+        monthMaxDate = date.getMonth().maxLength();
+        calendar.getChildren().clear();
+        drawCalendar();
+    }
+
+    @FXML
+    void jumpToNow() {
+        date = ZonedDateTime.now();
+        monthMaxDate = date.getMonth().maxLength();
+        calendar.getChildren().clear();
+        drawCalendar();
+    }
+
+    private void refreshCalendar() {
+        calendar.getChildren().clear();
+        drawCalendar();
+    }
+
+    private void drawCalendar() {
+        year.setText(String.valueOf(date.getYear()));
+        month.setText(String.valueOf(date.getMonth()));
+
+        // List of activities for a given month
+        Map<Integer, List<Event>> eventsForMonthMap = getEventsForMonth();
+
+        // Check for leap year
+        if (isLeapYear()) {
+            monthMaxDate = 28;
+        }
+        drawCalenderBoxes(eventsForMonthMap);
+    }
+
+    private Map<Integer, List<Event>> getEventsForMonth() {
+        List<Event> events = new ArrayList<>();
+        eventList.forEach(e -> {
+            if (isInMonth(e)) {
+                events.add(e);
+            }
+        });
+        return getCalenderMap(events);
+    }
+
+    private Map<Integer, List<Event>> getCalenderMap(List<Event> events) {
+        Map<Integer, List<Event>> calendarEventsMap = new HashMap<>();
+        for (Event e : events) {
+            int date = e.getDate().getDay();
+            if (!calendarEventsMap.containsKey(date)) {
+                calendarEventsMap.put(date, List.of(e));
+            } else {
+                List<Event> oldList = calendarEventsMap.get(date);
+                List<Event> newList = new ArrayList<>(oldList);
+                newList.add(e);
+                calendarEventsMap.put(date, newList);
+            }
+        }
+        return calendarEventsMap;
+    }
+
+    private void drawCalenderBoxes(Map<Integer, List<Event>> eventsForMonthMap) {
+        int dateOffset = ZonedDateTime.of(date.getYear(), date.getMonthValue(), 1, 0,
+                0, 0, 0, date.getZone()).getDayOfWeek().getValue() % 7;
+
+        for (int i = 0; i < 6; i++) {
+            for (int j = 0; j < 7; j++) {
```

Comment:
In terms of code quality, should refrain from using magic numbers 6 and 7 in the two for-loops. 
Perhaps initialising these numbers may improve code readability?

Reply:
Good idea